# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/gc/bug-1648901.js
# Commit: ac6d26908599
# Full Hash: ac6d26908599e15b1c129cbb0f252ca9b82836c6
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-07-03 03:56:55
# Description:
#   Bug 1648901 - Fix crash tracing partially constructed FinalizationRegistrationsObject r=sfink
#   
#   The shell allocation metadata hook means we can see these before they've been fully initialised.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D81521
# ==============================================================================

diff -r 21569aa4bced -r ac6d26908599 js/src/jit-test/tests/gc/bug-1648901.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/gc/bug-1648901.js	Tue Jun 30 22:31:29 2020 +0000
@@ -0,0 +1,7 @@
+// |jit-test| skip-if: !('gczeal' in this)
+
+gczeal(15);
+enableShellAllocationMetadataBuilder();
+var registry = new FinalizationRegistry(x => 0);
+gczeal(9, 3);
+registry.register({}, 1, {});
